tests/test_rois.py

```diff
@@ -1,209 +1,257 @@
-                           [1, 2, 3, 1, 2, 3, 1, 2, 3],
-                           [0.75, 0.89, 0.92, 1, 0.43, 0.56, 0.98, 0.3, 0.1],
-                           (5, 5), 1, 1, 0.5, 2, np.array([[1, 1, 1, 1, 1],
-                                                           [1, 1, 1, 1, 1],
-                                                           [1, 1, 0, 1, 1],
-                                                           [1, 1, 0, 1, 1],
-                                                           [1, 1, 1, 0, 0]]))
-                          ])
-def test_edge_dilated_mask(coo_rows, coo_cols, coo_data, video_shape,
-                           segmentation_id, roi_id, threshold, stroke_size,
-                           expected_mask):
-    """
-    Test Cases:
-        1) 3 x 3 object, dilated once to removes middle
-        2) 3 x 3 object with overhang at row 2 col 0, dilated once removes all but edges
-        3) 2 x 2 object centered in the bottom right quadrant, dilated once doubles width and height
-        4) 3 x 3 object centered in the middle of frame with hole at center, dilated twice
-        5) 0 x 0 object, dilated three times maintains no valid pixels
-        6) 1 x 1 object in 3 x 3 frame, dilated twice to fill frame
-        7) Two 2 x 2 objects in 5 x 5 frame, disjoint and not touching, dilated once
-        8) 3 x 3 object with certain pixels below threshold, dilated twice
+@pytest.mark.parametrize(
+        ("weighted", "expected", "athresh", "quantile", "full"),
+        [
+            (
+                np.array([
+                    [0.0, 0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.5, 1.0, 0.0, 0.0],
+                    [0.0, 2.0, 2.0, 1.5, 0.0],
+                    [2.0, 1.0, 1.5, 0.5, 0.0],
+                    [0.0, 0.0, 0.0, 0.0, 0.0]
+                    ]),
+                np.array([
+                    [0, 0, 0, 0, 0],
+                    [0, 0, 1, 0, 0],
+                    [0, 1, 0, 1, 0],
+                    [1, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0]
+                    ]),
+                0.7,
+                None,
+                True),
+            (
+                np.array([
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 0.0, 0.5, 0.0, 0.0, 1.0],
+                    [1.0, 0.0, 2.0, 1.0, 0.5, 1.0],
+                    [2.0, 0.0, 1.0, 1.5, 0.5, 1.0],
+                    [1.0, 0.0, 0.0, 0.0, 0.0, 1.0],
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
+                    ]),
+                np.array([
+                    [1, 1, 1, 1, 1, 1],
+                    [1, 0, 0, 0, 0, 1],
+                    [1, 0, 1, 1, 0, 1],
+                    [1, 0, 1, 1, 0, 1],
+                    [1, 0, 0, 0, 0, 1],
+                    [1, 1, 1, 1, 1, 1]
+                    ]),
+                0.7,
+                None,
+                True),
+            ])
+def test_roi_generate_outline(weighted, full, expected, athresh, quantile):
```

Comment:
I think as with above comment, adding descriptions for these test cases would be helpful.

Reply:
added